src/SdfGenerator.cc

```diff
@@ -422,6 +430,67 @@ namespace sdf_generator
     return true;
   }
 
+  /////////////////////////////////////////////////
+  /// \brief Checks if a string is a number
+  /// \param[in] _str The string to check
+  /// \return True if the string is a number
+  bool isNumber(const std::string &_str)
+  {
+    for (const char &c : _str)
+      if (!std::isdigit(c)) return false;
+
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  void updateModelElementWithNestedInclude(sdf::ElementPtr &_elem,
+                                           const bool _saveFuelVersion,
+                                           const IncludeUriMap &_includeUriMap)
+  {
+    sdf::ElementPtr e = _elem->GetFirstElement(), nextE = nullptr;
+    while (e != nullptr)
+    {
+      nextE = e->GetNextElement();
+
+      if (e->GetIncludeElement() != nullptr)
+      {
+        std::string modelDir = common::parentPath(e->FilePath());
+        auto uriMapIt = _includeUriMap.find(modelDir);
+
+        if (_saveFuelVersion && uriMapIt != _includeUriMap.end())
+        {
+          // find fuel model version from file path
+          std::string version = common::parentPath(e->FilePath());
+          version = common::basename(version);
+
+          if (isNumber(version))
+          {
+            sdf::ElementPtr uriElem = e->GetIncludeElement()->GetElement("uri");
+            std::string uri = uriElem->GetValue()->GetAsString();
+            uri = uri + "/" + version;
+            uriElem->Set(uri);
+          }
+          else
+          {
+            ignwarn << "Error retrieving Fuel model version,"
+                    << " saving model without version."
+                    << std::endl;
+          }
+        }
+
+        _elem->InsertElement(e->GetIncludeElement());
+        e->RemoveFromParent();
```

Comment:
I noticed another thing. This changes the order of models. I'm testing with `model_nested_include.sdf`  and the original has:
```xml
      <model name="M2">
        <include>
          <uri>sphere</uri>
          <pose>0 2 2 0 0 0</pose>
        </include>

        <model name="M3">
          <include>
            <name>coke</name>
            <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Coke Can</uri>
            <pose>2 2 2 0 0 0</pose>
          </include>
        </model>
      </model>
```
The saved file has
```xml
      <model name='M2'>
        <model name='M3'>
          <include>
            <name>coke</name>
            <uri>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Coke Can</uri>
            <pose>2 2 2 0 -0 0</pose>
          </include>
        </model>
        <include>
          <uri>sphere</uri>
          <pose>0 2 2 0 -0 0</pose>
        </include>
      </model>
```
(with `<plugin>` removed). The order in the original is `sphere`, `M3`, whereas in the saved file `M3`,`sphere`. The two files will have different results when `sdf::Model::ModelByIndex` is called. I think can preserve the order if we do this:

```suggestion
        e ->Copy(e->GetIncludeElement());
```

This is the first time I've used `sdf::Element::Copy` and I now see its usefulness.

Reply:
Thanks for noticing that! 73b71101702286e5161854ec650ff9aa39c0ed0d

Reply:
I'm now getting the `//include/@name attribute set.
```
      <include name='include_nested_new_name'>
        <uri>include_nested</uri>
        <name>include_nested_new_name</name>
        <pose>1 2 3 0 -0 0</pose>
      </include>
```
I believe it's because `e` is a model and has a `name` attribute and `sdf::Element::Copy` doesn't seem to be clearing the attributes when copying. I think this is something we should fix in libsdformat. Since I don't know the reason for why attributes are not cleared in https://github.com/osrf/sdformat/blob/25004f0bad227cf11854cc208e74be08e8623359/src/Element.cc#L218-L220, I think it might be best to create another function to clear attributes instead of modifying `sdf::Element::Copy`.

Reply:
Should we create both a `sdf::Element::ClearAttributes()` as well as `sdf::Element::RemoveAttribute(string key)`? Just to have both options for users

Reply:
Yeah, that sounds good to me.

Reply:
Here's the PR that adds these methods: https://github.com/osrf/sdformat/pull/555

This will be forward ported to dome & edifice once merged